ctags/parsers/c.c

```diff
@@ -3133,18 +3132,22 @@ static void createTags (const unsigned int nestLevel,
 	DebugStatement ( if (nestLevel > 0) debugParseNest (false, nestLevel - 1); )
 }
 
-static bool findCTags (const unsigned int passCount)
+static rescanReason findCTags (const unsigned int passCount)
 {
 	exception_t exception;
-	bool retry;
+	rescanReason rescan = RESCAN_NONE;
+	kindOption *kind_for_define = NULL;
 
 	contextual_fake_count = 0;
 
 	Assert (passCount < 3);
-	cppInit ((bool) (passCount > 1), isInputLanguage (Lang_csharp), isInputLanguage (Lang_cpp), &(CKinds [CK_DEFINE]));
+
+	kind_for_define = CKinds+CK_DEFINE;
+
+	cppInit ((bool) (passCount > 1), isInputLanguage (Lang_csharp), isInputLanguage(Lang_cpp),
+		kind_for_define);
 
 	exception = (exception_t) setjmp (Exception);
-	retry = false;
```

Comment:
shouldn't `rescan` be reset to `RESCAN_NONE` here?  `setjmp()` might lead to jumping there directly.